integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/projection/IdProjectionIT.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.pojo.projection;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.search.integrationtest.mapper.pojo.testsupport.util.rule.JavaBeanMappingSetupHelper;
+import org.hibernate.search.mapper.javabean.mapping.SearchMapping;
+import org.hibernate.search.mapper.javabean.session.SearchSession;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.GenericField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class IdProjectionIT {
+
+	@Rule
+	public BackendMock backendMock = new BackendMock();
+
+	@Rule
+	public JavaBeanMappingSetupHelper setupHelper = JavaBeanMappingSetupHelper.withBackendMock(
+			MethodHandles.lookup(), backendMock );
+
+	private SearchMapping mapping;
+
+	@Before
+	public void setup() {
+		backendMock.expectSchema( IndexedEntity.INDEX, b -> b.field( "text", String.class ) );
+		mapping = setupHelper.start().setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void select_searchProjection_id_wrongIdentifierType() {
+		try ( SearchSession session = mapping.createSession() ) {
+			assertThatThrownBy( () -> session.search( IndexedEntity.class )
+					.select( f -> f.id( String.class ) )
+			)
+					.isInstanceOf( SearchException.class )
+					.hasMessageContainingAll(
+							"The required identifier type 'java.lang.String' does not match the actual identifier type 'java.lang.Integer'"
+					);
+		}
+	}
```

Comment:
I think you can remove this.

On the backend side, you're already testing invalid projection types in `IdentifierSearchProjectionSingleValuedIT#invalidProjectionType`.

On the mapper side, it does indeed make sense to test this too, but IMO the only reason it makes sense is because we have many different bridges with different types. So I'd suggest you test this for every default bridge.

Fortunately, there's an easy way to do it: add a test method in `DocumentIdDefaultBridgeBaseIT` instead. You can take inspiration from `org.hibernate.search.integrationtest.mapper.pojo.mapping.definition.FieldDefaultBridgeBaseIT#indexToProjectionConverter`.